File: app/src/main/java/quangnam/com/sample/di/PerChildFragment.java
package quangnam.com.sample.di;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

/**
 * Created by quangnam on 1/13/18.
 * Project Sample
 */

@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface PerChildFragment {
}

File: app/src/main/java/quangnam/com/sample/di/PerChildFragment.kt
package quangnam.com.sample.di

import java.lang.annotation.RetentionPolicy
import javax.inject.Scope

/**
 * Created by quangnam on 1/13/18.
 * Project Sample
 */

@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class PerChildFragment
